coalib/results/Diff.py

```diff
@@ -351,14 +351,22 @@ def __bool__(self):
     def delete_line(self, line_nr):
         """
         Mark the given line nr as deleted. The first line is line number 1.
+
+        Raises an exception if line nr doesn't exist in the diff.
         """
-        linediff = self._get_change(line_nr)
-        linediff.delete = True
-        self._changes[line_nr] = linediff
+        if line_nr > len(self._file):
+            raise ValueError('The given line number is out of bounds')
+        else:
```

Comment:
There's no need for an else here. If an exception is raised, control would automatically go back to the calling function. Leave a line break.